src/main/scala/org/camunda/feel/impl/builtin/ConversionBuiltinFunctions.scala

```diff
@@ -169,14 +154,17 @@ class ConversionBuiltinFunctions(valueMapper: ValueMapper) {
   private def numberFunction =
     builtinFunction(
       params = List("from"),
-      invoke = { case List(ValString(from)) =>
-        ValNumber(from)
+      invoke = {
+        case List(ValString(from)) if !isNumeric(from) => ValNull
```

Comment:
:wrench: Instead of returning `ValNull`, we could return a `ValError()` with a proper failure message. As a result, the engine would use `null` but also produce a warning for debugging purposes.

We can verify the warning in the test case. See the example in [ComparisonTypeTest.scala](https://github.com/camunda/feel-scala/blob/main/src/test/scala/org/camunda/feel/impl/interpreter/ComparisonTypeTest.scala#L51-L55).